### PR TITLE
Search results page

### LINKED ISSUE #43: Create a default header.php file
**As a page template, I should call a theme-wide `header.php` file instead of using my own declarations for header.**
AC1: Each page template should call `header.php`.
AC2: Any legacy items that `header.php` already controls should be removed from page templates.

### PR DESCRIPTION
**Pull request by miina**
This PR is based on #42.
Fixes #43.

Adds search results page with dynamic data using amp live list.

Todo:
- Styling might need fixes;

screen shot 2018-04-18 at 8 33 34 pm

### CODE REVIEW

**Comment by miina** [comment]
@mehigh Finished the template for now, the search CSS can be found from assets/css/search.css, likely that there is something that needs fixing. Also, there is no search button, should we create this (cc @kienstra)? Or how should the search work?

**Comment by kienstra** [comment]
**Proposed Search Button**

Hi @postphotos,
Could you please help with @miina's question about the search button?

>Also, there is no search button, should we create this (cc @kienstra)? Or how should the search work?

The [design](https://www.figma.com/file/L14jTE7iNGOE7DH57Ro7ncyu/Travel-Theme?node-id=2%3A118) doesn't look to have a search button on the search results page:

no-search-button

And neither does the [AMP Start theme on Firebase](https://ampstart-iframes.firebaseapp.com/templates/travel/travel-results.amp). What do you think of this possible "search" button that I created?

possible-search-button

The button is a few pixels too tall, though.

```html
<button class="ampstart-btn rounded bold">Find Adventures</button>
```

Styling (though this wouldn't be added inline in the actual PR):
```css
element.style {
    background: rgba(0,0,0,.2);
    margin-left: 25px;
}
```

**Comment by kienstra** [comment]
This applies to Issue #29.

**Comment by miina** [comment]
@kienstra This should also be ready for review now, added the search button, too. Note that on the mobile view the inputs and filters are removed and tried to make something similar to the [static theme's view](https://ampstart-iframes.firebaseapp.com/templates/travel/travel-results.amp).
One more note -- currently there is no design / template for no results page, added just a text. Could this stay like that? Or should we create something better?

@mehigh Would be great if you could also review the template, specifically for HTML / CSS adjustments.

**Comment by miina** [comment]
Hey @kienstra, I think that this should be ready for review as well. Added a button for searching, let me know if that seems to look fine.

I'll ask @mehigh separately to take a look from HTML / CSS perspective.

File: assets/css/search.css [hunk 1]
```diff
@@ -2,7 +2,18 @@
...
...ted] .travel-type-selector-active,.travel-type-selector [option][selected]:hover .travel-type-selector-active{opacity:1;-webkit-clip-path:circle(142% at 0 50%);clip-path:circle(142% at 0 50%)}.travel-type-selector [option][selected]:hover .travel-type-selector-active{opacity:1;-webkit-clip-path:circle(calc(142% - 36px) at 0 50%);clip-path:circle(calc(142% - 36px) at 0 50%)}.travel-type-selector-active .travel-type-selector-content,.travel-type-selector-active:after{color:#fff}.travel-type-selector [option][selected]{outline:none;cursor:pointer}.travel-type-selector [option]:hover .travel-type-selector-active:after,.travel-type-selector [option]:hover .travel-type-selector-inactive:after,.travel-type-selector [option][selected] .travel-type-selector-active:after,.travel-type-selector [option][selected] .travel-type-selector-inactive:after{-webkit-transform:rotate(45deg);transform:rotate(45deg)}.travel-type-selector [option][selected]:hover:not(:focus) .travel-type-selector-active:after,.travel-type-selector [option][selected]:hover:not(:focus) .travel-type-selector-inactive:after{-webkit-transform:rotate(0deg);transform:rotate(0deg)}.travel-type-selector [option]:focus{outline:none}
```
Comment (kienstra) [inline assets/css/search.css:3]: It looks like this is stylesheet is appearing correctly in the `<stye amp-custom>`. But there look to be some styling issues:

search-php-styling

File: templates/entry/slim.php [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+	<div class="relative travel-results-result">
+		<a class="travel-results-result-link" href="<?php the_permalink(); ?>">
+			<amp-img class="rounded bg-silver mb1" width="2" height="1" noloading="" layout="responsive" src="<?php echo esc_url( wp_get_attachment_url( get_post_thumbnail_id() ) ); ?>" ></amp-img>
+		</a>
+	</div>
+	<div class="h2 line-height-2 mb1">
+		<span class="travel-results-result-text"><?php echo get_the_title(); ?></span>
+		<span class="travel-results-result-subtext h3">&bull;</span>
+		<span class="travel-results-result-subtext h3">$&nbsp;</span>
+		<span class="bold"><?php echo esc_html( get_post_meta( get_the_ID(), 'amp_travel_price', true ) ); ?></span>
```
Comment (kienstra) [inline templates/entry/slim.php]: What do you think about outputting the 3 lines above only if the `'adventure'` has a price?

File: templates/entry/slim.php [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+	</div>
+	<?php get_template_part( 'templates/entry/meta/slim-meta' ); ?>
+</div>
```
Comment (kienstra) [inline templates/entry/slim.php]: It looks like the `$` and price sometimes appear on a different line:

styling-product

The [Firebase demo](https://ampstart-iframes.firebaseapp.com/templates/travel/travel-results.amp) doesn't look to have this issue with the `$`. But it is possible for the `*` and title to be on the same line, like for "Bake Traditional Sweet Bread."

In my opinion, the most important thing would be to not have the `$` and price on separate lines, and the bullet appearing on a separate line isn't an issue. But I'm happy to defer to @postphotos for this.

File: templates/entry/meta/slim-meta.php [hunk 4]
```diff
@@ -0,0 +1,22 @@
...
+$reviews   = wp_count_comments( get_the_ID() );
+?>
+<div class="h4 line-height-2">
+	<?php get_template_part( 'templates/entry/meta/slim-rating' ); ?>
+	<?php /* translators: %d: Number of reviews. */ ?>
+	<span class="travel-results-result-subtext mr1"><?php printf( esc_html__( '%d Reviews', 'travel' ), esc_html( $reviews->approved ) ); ?></span>
+	<span class="nowrap slim-location">
+		<span class="travel-icon travel-img-icon-map-pin-outline-gray"></span>
+		<?php echo esc_html( $location ); ?>
+	</span>
```
Comment (kienstra) [inline templates/entry/meta/slim-meta.php]: What do you think about outputting these 4 lines above only if there is a a location for the `'adventure'`?

File: templates/entry/meta/slim-meta.php [hunk 5]
```diff
@@ -0,0 +1,22 @@
...
+$locations = wp_get_post_terms( get_the_ID(), 'location', array(
+	'fields' => 'names',
+) );
+$location  = ! empty( $locations ) ? $locations[0] : '--';
+$reviews   = wp_count_comments( get_the_ID() );
+?>
+<div class="h4 line-height-2">
+	<?php get_template_part( 'templates/entry/meta/slim-rating' ); ?>
+	<?php /* translators: %d: Number of reviews. */ ?>
+	<span class="travel-results-result-subtext mr1"><?php printf( esc_html__( '%d Reviews', 'travel' ), esc_html( $reviews->approved ) ); ?></span>
```
Comment (kienstra) [inline templates/entry/meta/slim-meta.php]: How about:
```php
esc_html( _n( '%d Review', '%d Reviews', $reviews->approved, 'travel' ) )
```
in place of:
```php
esc_html__( '%d Reviews', 'travel' )
```

Though it's not part of this PR's diff, could you please make a similar change to [functions.php](https://github.com/xwp/travel/blob/1a4375af8bfb3feeffcf5a5aeee5ae1465901361/includes/functions.php#L126)?

File: templates/navbar-search.php [hunk 6]
```diff
@@ -3,69 +3,70 @@
...
+				</a>
+				<div class="search-info flex flex-auto items-center md-hide lg-hide">
+					<div>
+						<div class="h3 line-height-2 bold">
+							<?php esc_html_e( 'Results for:', 'travel' ); ?>
+						</div>
+						<div class="travel-results-navbar-subheading h6 caps line-height-4">
+							<?php echo ! empty( $_GET['s'] ) ? esc_attr( $_GET['s'] ) . '•' : ''; ?>
+							<?php echo ! empty( $_GET['start'] ) ? sprintf( 'From %s', esc_attr( $_GET['start'] ) ) : ''; ?>
+							<?php echo ! empty( $_GET['end'] ) ? sprintf( 'Until %s', esc_attr( $_GET['end'] ) ) : ''; ?>
```
Comment (kienstra) [inline templates/navbar-search.php]: Could you please have these 3 lines above use `esc_html()` instead of `esc_attr()`? The result is the same for the normal values, though.

File: search.php [hunk 7]
```diff
@@ -1,55 +1,43 @@
...
-                    else: ?>
-                    <div>No results</div>
-                    <?php endif; ?>
-                </div>
-            </div>
-        </section>
-        <!--/ Results -->
-    </div>
+		<!-- Results -->
+		<section class="travel-results pb1 md-pt1">
```
Comment (kienstra) [inline search.php:14]: It sounds like the `taxonomy.php` or `taxonomy-*.php` template will be mainly the same as this `search.php`. 

So it might be good to extract this `section.travel-results` into a separate template. But we can do that in #37.

File: templates/navbar-search.php [hunk 8]
```diff
@@ -3,69 +3,70 @@
...
+									fields_end: event.value
+								})' title="yyyy-mm-dd" name="return"
 								value="<?php echo isset( $_GET['end'] ) ? esc_attr( $_GET['end'] ) : ''; ?>">
 
 							<div class="travel-date-input-label">
-								<?php echo esc_html( 'Return', 'travel' ); ?>
+								<?php esc_html_e( 'Return', 'travel' ); ?>
 							</div>
 						</label>
+						<a href="<?php echo esc_url( site_url() ); ?>?s=<?php echo esc_attr( $_GET['s'] ); ?>" [href]="'<?php echo esc_url( site_url() ); ?>?s=' + fields_query_live + '&start=' + (fields_start || '') + '&end=' + (fields_end || '')" class="search-travel-btn travel-input travel-input-dark rounded ampstart-btn white bold">
```
Comment (kienstra) [inline templates/navbar-search.php]: It looks like entering date value(s) removes the text from the query. For example, in the screenshot below, the query doesn't include the text:
```
https://local.them/?s=&start=2018-04-02&end=2018-04-27
```

date-values-query

**Review by kienstra — CHANGES_REQUESTED** [review]
**Looks Good**
*Some Points Including Styling*

Hi @miina,
This looks good. Like you mentioned, it'd be great to have @mehigh's help with some styling issues mentioned below.

**Comment by kienstra** [comment]
**Question About "No Results"**

Hi @postphotos,
Could you please help with @miina's question:
>One more note -- currently there is no design / template for no results page, added just a text. Could this stay like that? Or should we create something better?

Here's how that looks at the moment:

no-results

**Comment by kienstra** [comment]
**Request For Help With Styling**

Hi @mehigh,
Are you available to help with some styling points [here](https://github.com/xwp/travel/pull/45#discussion_r182865234) and [here](https://github.com/xwp/travel/pull/45#discussion_r182871049)? Thanks for all of your work on this theme.

File: search.php (hunk 7 above)
Comment (miina) [reply search.php:14]: Thanks for noting that, will add it with #37.

File: templates/navbar-search.php (hunk 8 above)
Comment (miina) [reply templates/navbar-search.php]: Thanks for catching that!

**Comment by miina** [comment]
Hey @kienstra, thanks again for reviewing, the issues that are not related to styling should be fixed now.

File: templates/entry/slim.php (hunk 2 above)
Comment (kienstra) [reply templates/entry/slim.php]: Nice use of [entry/meta/price.php](https://github.com/xwp/travel/pull/45/commits/6e69f19c8e962a92c3777c67a7ea5c77ad03f8df#diff-a044d91ea4383c8b07d6cb9d97b277fcR1) for this.

File: templates/entry/meta/slim-meta.php (hunk 5 above)
Comment (kienstra) [reply templates/entry/meta/slim-meta.php]: Thanks for making these changes!

**Comment by kienstra** [comment]
**Thanks, PR Approved Other Than Styling**

 Hi @miina,
Thanks for addressing those points. This is approved, other than the styling points [here](https://github.com/xwp/travel/pull/45#discussion_r182865234) and [here](https://github.com/xwp/travel/pull/45#discussion_r182871049).

File: assets/css/search.css (hunk 1 above)
Comment (mehigh) [reply assets/css/search.css:3]: I've fixed this by removing the inline left float and by adding the flex wrapper @miina.

**Comment by mehigh** [comment]
I've identified the reason the images look squashed. In the demo all of the images had a width of 2 and a height of 1, which corresponded to a 2:1 format image. Since the images in your screenshot were not 2:1 that didn't work. I'm switching this to use the correct aspect ratio by reading the width & height from the thumbnail meta.

**Comment by mehigh** [comment]
The reason the "$" wasn't appearing alone in the demo theme https://www.ampstart.com/templates/travel/travel-results.amp was the fact that after the span containing the $ there was no whitespace or newline. The fix for this was simply removing the whitespace altogether. There was correctly a non-breakable space in there, it only needed to be alone in order for the effect to work correctly.

**Comment by mehigh** [comment]
@kienstra this should be good to go now, as i've handled all 3 styling issues reported - price wrapping on a separate line than the $, the grid and the image formats.

**Review by mehigh — APPROVED** [review]
All looks good to me.

**Comment by kienstra** [comment]
**Thanks!**

Hi @mehigh,
Thanks a lot for fixing the display of the images, adding a flex wrapper, and making the `$` always appear next to the price. This looks good.

**Comment by kienstra** [comment]
**New Image Size To Account For 640px Display**

If it's alright, I [added a new image size](https://github.com/xwp/travel/pull/45/commits/997e836aae8db3739097408cd9177ead392fd09c) of `600x300`. When the viewport is 640px, the image is about 600px:

image-roughly-600-300

Before, the images had a lower resolution at that size. This keeps the expected 2-1 ratio.

In testing this, it helped me to use the WP-CLI comand:
```bash
wp media regenerate
```

**Review by kienstra — APPROVED** [review]
